### PR TITLE
Game state refactor

### PR DESCRIPTION
**Pull request by dehli**
# PR for [#344 (API)](https://github.com/willowtreeapps/wombats-api/pull/344).

## PR Status
**READY/IN DEVELOPMENT/HOLD**

## Ready for PR?
- [X] I have updated the CHANGELOG with an entry including a description, a link to my profile, and a link to the issue ticket. This change is filed under an Enhancement or Bug Fix, is shown under the relevant release tag name, and is included in my PR branch.
- [X] I have run `lein run-lint` to check for linting errors.

## Changes Proposed:
- Update app so it works with game state refactor.
- Made it so players are live sorted based on score.

## Breaking changes?
- Hopefully none.

@emilyseibert @oconn @dehli

### CODE REVIEW

File: src/cljs/wombats_web_client/components/game_ranking.cljs [hunk 1]
```diff
@@ -18,33 +18,48 @@
...
-                                    (get-adjusted-hp game)
-                                    (get-bounded-hp hp))
-                is-dead (zero? adjusted-hp-value)
-                health-color (get-health-color adjusted-hp-value)
-                health-percent (str adjusted-hp-value "%")]
-    ^{:key (or username id)}
+(defn- get-player-hp
+  [game player-color]
+  (let [arena (get-in game [:game/frame :frame/arena])
+        wombats (filter #(let [contents (:contents %)]
```
Comment (emilyseibert) [inline src/cljs/wombats_web_client/components/game_ranking.cljs]: util function or separate function

File: src/cljs/wombats_web_client/panels/game_play.cljs [hunk 2]
```diff
@@ -103,49 +113,34 @@
...
+      (show-winner-modal players))
 
     [:div.right-game-play-panel
 
      [:div.top-panel
-      [game-play-title game (zero? user-bots-count) game-id]
-      [game-play-subtitle game]
-      [max-players game]
+      [game-play-title game (not in-game)]
+      [:h2.game-play-subtitle (str name " - High Score")]
```
Comment (dehli) [inline src/cljs/wombats_web_client/panels/game_play.cljs]: Gonna change this `- High Score`

File: src/cljs/wombats_web_client/components/game_ranking.cljs [hunk 3]
```diff
@@ -18,33 +18,48 @@
...
-  [{:keys [game/stats] :as game}]
-  [:div.game-ranking-box
-   [:ul.list-wombat-status
-    (doall (map #(render-wombat-status game %) stats))]])
+  [game]
+  (let [game @game]
+    [:div.game-ranking-box
+     [:ul.list-wombat-status
+      (doall (map #(render-wombat-status game %)
+                  (:game/players game)))]]))
```
Comment (emilyseibert) [inline src/cljs/wombats_web_client/components/game_ranking.cljs]: variable in let

File: src/cljs/wombats_web_client/panels/game_play.cljs [hunk 4]
```diff
@@ -103,49 +113,34 @@
...
+      [:p.wombat-counter
+       (str "Wombats: " (count players) "/" max-players)]
```
Comment (emilyseibert) [inline src/cljs/wombats_web_client/panels/game_play.cljs]: https://zpl.io/cS2Ef

can you add in rounds here?

File: src/cljs/wombats_web_client/utils/games.cljs [hunk 5]
```diff
@@ -83,3 +83,20 @@
                         (user-in-game? current-user game))
                  (conj coll game)
                  coll)) [] games))
+
+(defn get-player-by-username
+  [username players]
+  (first (filter #(= (get-in %
```
Comment (emilyseibert) [inline src/cljs/wombats_web_client/utils/games.cljs]: filter on new line

**Pull request merged**